### ISSUE TITLE
Add OnDevice Markup Extension

### ISSUE THREAD

**Issue by michael-hawker** [unoplatform/Uno.WindowsCommunityToolkit#60]
## I'm submitting a...
Once Uno supports Markup Extension with https://github.com/unoplatform/uno/issues/39, it'd be cool if the OnDevice one could support other platforms as well from Uno. 😊 (Android, iOS, WASM)

  
- Feature request

## Current behavior
https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/master/Microsoft.Toolkit.Uwp.UI/Extensions/Markup/OnDevice.cs

https://github.com/unoplatform/uno/issues/39

**Comment by Massimo37** [comment]
Unfortunately this still won't work in Uno due to the fact that when I process a custom MarkupExtension, I expect a MarkupExtensionReturnType to be specified. I would need to discuss this with @davidjohnoliver

**Comment by michael-hawker** [comment]
@Massimo37 could we just put a `MarkupExtensionReturnType(typeof(object))` on there?

**Comment by Massimo37** [comment]
@michael-hawker yes I thought about that, it's an option I'd like to try once the PR goes through. It would be versatile if the MarkupExtensionReturnType is optional and allows a dev to return different entities (as is allowed with OnDevice).

**Comment by Massimo37** [comment]
This show now be possible to do, I've made the MarkupExtensionReturnType  optional 👍

**Comment by Massimo37** [comment]
*MarkupExtensionReturnType* is now optional, so that part is done. Unfortunately we still need this feature to be completed until this markup extension is usable:

https://github.com/unoplatform/uno/issues/1468

**Comment by MartinZikmund** [comment]
I'm on it :-)

**Comment by MartinZikmund** [comment]
https://github.com/unoplatform/uno/pull/1970
